### PR TITLE
Refactor live pricing

### PR DESCRIPTION
**Pull request by shaundon**
Aims of this refactor:
* Get rid of that global state object.
* Make it more readable / understandable.

Other things to note:
* Removed the timeout period and retries, it felt a bit YAGNI.
* If it errors, it now sends a code 500 with the error to the front end. 
* Removed `lodash` dependency.

### CODE REVIEW

File: server/src/server.js [hunk 1]
```diff
@@ -28,6 +28,12 @@ app.get('/api/search', (req, res) => {
   livePricing.search({
     // TODO client to provide params
     // check in api docs what client should provide
+    adults: 1,
+    class: 'Economy',
+    toPlace: 'EDI-sky',
+    fromPlace: 'LOND-sky',
+    fromDate: '2018-08-20',
+    toDate: '2018-08-30',
```
Comment (shaundon) [inline server/src/server.js]: Debug code, should be removed before merge.

File: server/src/server.js [hunk 2]
```diff
@@ -19,22 +19,24 @@ app.get('/', (req, res) => {
 /**
   Simple flight search api wrapper.
 
-  TODO: client should provide params
+  TODO: client should provide params.
```
Comment (arnaugm) [inline server/src/server.js:22]: I like the level of commitment to improve user experience demonstrated here 😂

**Comment by arnaugm** [comment]
LGTM

**Review by arnaugm — APPROVED** [review]

File: server/src/server.js (hunk 2 above)
Comment (shaundon) [reply server/src/server.js:22]: I'm a bit of a stickler for correct grammar in user-facing content 😂

**Review by matteo-hertel — APPROVED** [review]
hero!

**Pull request merged**